source/codegen/templates/service_helpers.mako

```diff
@@ -258,6 +270,12 @@ one_of_case_prefix = f'{namespace_prefix}{function_name}Request::{PascalFieldNam
         response->set_${parameter_name}(${parameter_name});
 %     elif common_helpers.is_struct(parameter):
         Copy(${parameter_name}, response->mutable_${parameter_name}());
+%     elif parameter['type'] == 'ViBoolean[]':
+        int i = 0;
+        for (auto item : ${parameter_name}) {
+          ${parameter_name}_mutable_data[i] = item;
+          i++;
+        }
```

Comment:
See my comment on the .cpp file. You should remove this and add `parameter['type'] == 'ViBoolean[]'` to the previous struct case. In fact, this shouldn't be specific to just `ViBoolean`. It's true of any non-32-bit numeric type. Add a `common_helpers.is_non32bit_integer` helper.

Reply:
I merged it with `struct` case above, but couldn't generalize it to non-32-bit integers. My `Copy` function is tied to ViBoolean and bool types. Also, apart from ViBoolean, I believe only other non-32-bit integer we have is ViInt16 (?) and codegen for arrays of that type is disabled right now. See my comment on `service.cpp.mako`.